pylint/checkers/utils.py

```diff
@@ -852,3 +852,20 @@ def is_registered_in_singledispatch_function(node):
             return decorated_with(func_def, singledispatch_qnames)
 
     return False
+
+
+def infer_sequence_with_starred_expressions(seq_node):
```

Comment:
What we can do is trying a best effort approach. We can have an inference function for Starred nodes, which yields its unpacked values, if inferrable, otherwise returns Uninferable. We will still need a helper in pylint, for handling the Starred explicit inference, since inferring a tuple with starred nodes in it will not automatically unpack the elts. I think having a bit more improved inference and having this bug fixing with a that and helper is a win win situation.